beancount/parser/printer_test.py

```diff
@@ -89,15 +89,17 @@ def assertRoundTripViaRealFile(self, entries1, errors1, test_write_source=True):
         # Contex manager can be used as of python 3.12
         # see https://github.com/python/cpython/issues/58451
         # https://github.com/beancount/beancount/issues/222
-        oss1 = tempfile.NamedTemporaryFile("w", suffix=".beancount", delete=False)
+        oss1 = tempfile.NamedTemporaryFile(
+            "w", suffix=".beancount", delete=False, newline="\n"
+        )
```

Comment:
I would use the native line ending for the platform. Instead of changing the newline style here, change the code where more lines are appended to use native newlines.

Reply:
we already use native newlines, add argument here is to get a platform-independent result in tests

Reply:
I'm not sure I understand. Where is the platform dependent value compared to a platform independent one?

Reply:
`oss2` here is a `io.StringIO` and doesn't do any newline handling, do we need to let `tempfile.NamedTemporaryFile` do not change any newlines.

Reply:
I find the reason, it's a but in escape_string

Reply:
looks like this is expected? `\n` is not escaped.

https://github.com/beancount/beancount/blob/07406fd58de27deb8920663af85659e0e2c81eba/beancount/utils/misc_utils_test.py#L236-L240

Reply:
it's a bug of parser, can't be fixed in this PR